File: 백준/Silver/18126. 너구리 구구/너구리 구구.py
import sys
sys.setrecursionlimit(1000000)

n = int(input())
graph = [[] for _ in range(n + 1)]

for i in range(n-1):
  a, b, c = map(int, input().split())
  graph[a].append((b, c))
  graph[b].append((a, c))

def dfs(visited, v, w):
  visited[v] = True
  max_w = 0
  
  for i in graph[v]:
    nv, nw = i

    if not visited[nv]:
      dw = dfs(visited, nv, w + nw)
      max_w = max(max_w, dw)

  return max(w, max_w)

visited = [False] * (n + 1)
print(dfs(visited, 1, 0))

File: 백준/Silver/18126. 너구리 구구/너구리 구구.cc
#include <bits/stdc++.h>
using namespace std;

vector<vector<pair<int, int>>> graph;
vector<bool> visited;
long long dist[5001] = {0};

void dfs(int v) {
	visited[v] = true;

	for (auto& e : graph[v]) {
		int nv = e.first;
		int nw = e.second;

		if (!visited[nv]) {
			dist[nv] = dist[v] + nw;
			dfs(nv);
		}
	}
}

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);

	int n;
	cin >> n;

	graph.resize(n+1);
	visited.resize(n+1, false);

	for (int i = 0; i < n-1; i++) {
		int a, b, c;
		cin >> a >> b >> c;
		graph[a].push_back({b, c});
		graph[b].push_back({a, c});
	}

	dfs(1);
	cout << *max_element(dist + 1, dist + n + 1);

	return 0;
}